Blog: An introduction to guardrail metrics (#6917)
PostHog/posthog.com: contents/blog/guardrail-metrics.md

As a developer, using guardrail metrics provides benefits like:

- More confidence A/B tests aren’t causing problems. You spend less time checking for the negative impact of an experiment.

- A more accurate source of issues than bug reports or general metrics. Developers don’t need to guess if an A/B test is causing an issue, guardrail metrics can tell them.

- More confidence in the results of the experiment knowing it doesn’t have broader negative effects.

- Promoting a culture of experimentation knowing A/B tests are safer to do. Creates trust between teams knowing other teams’ A/B tests won’t affect them.

### Why Airbnb uses guardrail metrics

[Airbnb](https://medium.com/airbnb-engineering/designing-experimentation-guardrails-ed6a976ec669) relies on guardrails to enable teams to run A/B tests while keeping the overall experience strong. They found A/B tests often positively impact one team’s metrics, while negatively impacting another. For example, if they didn’t display house rules, they saw an increase in bookings but a decrease in ratings.

> When a guardrail metric triggers, it escalates and stakeholders decide whether to continue the test or not. Guardrails at Airbnb flag ~25 experiments per month for review. 80% of these eventually launch after stakeholder discussion. They stop ~5 per month before launch, which means they prevent 5 potential negative impacts.

Author: Clarified on both

> ![](../images/blog/guardrail-metrics/trigger.png)

Author: Reworked

> ## What guardrail metrics should you pick?

Reviewer: ```suggestion
## How to choose guardrail metrics
```

As a general guideline, you should pick metrics important to the entire product or company. These are often [north star](/blog/north-star-metrics) or [product health](/blog/product-health-metrics) metrics. Specifically, Airbnb breaks guardrail metrics into three categories

- Key business metrics
- User experience metrics
- Strategic priority metrics

> It is important to have a mix of metrics from each category to ensure a broad enough coverage of potential issues. It is also important to not have too many metrics. Adding one increases the likelihood of false positives, which limits the throughput of experiments. For example, an [A/A test](/tutorials/aa-testing) with a significance level of 0.05 would expect to cause the following amount of false positives:
> 
> - With 3 guardrail metrics, you would have at least one false alert approximately 14% of the time.
> - With 10 guardrail metrics, approximately 40% of the time.
> - With 25 guardrail metrics, approximately 73% of the time.

Reviewer: ```suggestion
You need a mix of metrics from each category to ensure you don't miss potential issues. Just remember that each guardrail metric you add also increases the likelihood of false positives. 

For example, an [A/A test](/tutorials/aa-testing) with 3 guardrail metrics, and a significance level of 0.05, would cause a false alert approximately 14% of the time. The risk increases to 40% with 10 guardrail metrics, and a massive 73% with 25 guardrail metrics.

More false positives means you run fewer tests, limiting the pace of improvements. 
```

Could we recommend a sensible number of guardrail metrics per test?

Reviewer: This seems easier to read than the bullet points to me.

> ### Example guardrail metrics
> 
> For more insight into what specific metrics you should try, you can look at what other companies use as guardrail metrics. 
> 
> - Airbnb: revenue, bounce rate, page load speed, seats booked for experiences.
> 
> - [Square](https://developer.squareup.com/blog/lessons-learned-from-running-web-experiments/): bottom of funnel conversion, page load speed
> 
> - [Bing](https://exp-platform.com/Documents/2017-08%20KDDMetricInterpretationPitfalls.pdf): page load time, page returns
> 
> - [Netflix](https://www.adventuresinwhy.com/pdf/beyond_ab_testing.pdf): sample mismatch ratio, engagement, retention, conversion.
> 
> Other potential guardrail metrics include clickthrough rate, active users, customer lifetime value, and pageviews. 
> 
> > **Tip:** To help pick a metric, **i**t is useful to know what secondary metrics you have stopped experiments for in the past. These are the organization’s [revealed preference](https://en.wikipedia.org/wiki/Revealed_preference) for guardrail metrics and likely make good choices to formalize.

Reviewer: ```suggestion
### Example guardrail metrics

Potential guardrail metrics include clickthrough rate, active users, customer lifetime value, and pageviews. Here are some real-world examples:

- [Airbnb](https://medium.com/airbnb-engineering/designing-experimentation-guardrails-ed6a976ec669): revenue, bounce rate, page load speed, seats booked for experiences.

- [Square](https://developer.squareup.com/blog/lessons-learned-from-running-web-experiments/): bottom of funnel conversion, page load speed

- [Bing](https://exp-platform.com/Documents/2017-08%20KDDMetricInterpretationPitfalls.pdf): page load time, page returns

- [Netflix](https://www.adventuresinwhy.com/pdf/beyond_ab_testing.pdf): sample mismatch ratio, engagement, retention, conversion.

To help pick a metric, it's useful to know which secondary metrics have stopped experiments in the past. These are your company's [revealed preference](https://en.wikipedia.org/wiki/Revealed_preference) for guardrail metrics and likely make good choices to formalize.
```

> ## Your step-by-step guide to implementing guardrail metrics

Reviewer: ```suggestion
##  Implementing guardrail metrics in PostHog
```

Reviewer: I think we may as well just say PostHog rather than being open-ended with it.

Now that you know all about guardrail metrics, it’s time to implement them in your A/B tests. 

1. Create an experiment in PostHog.

2. Fill out the key name, variants, goal metric, and any other details. Adding details about the guardrail metrics to the description is useful.

3. Under "secondary metrics," click "Add metric," and set up the trend or funnel you want as a guardrail. 

4. Click "Save," implement the experiment in your app, and press "Launch."

> 5. Guardrail metrics display alongside your goal metrics on the experiment detail screen.

Reviewer: Would be nice to have a screenshot here, maybe showing where to see the secondary metrics being tracked?
